### PR TITLE
Read DR-bar parameters and pole masses from SLHA string

### PR DESCRIPTION
**Pull request by Expander**
Base the test on the parameters (DR-bar Lagrangian parameters, masses, ...) in the SLHA string.

(Before, only the input parameters in the SLHA string were used.)

### CODE REVIEW

**Comment by Expander** [comment]
BTW, I've updated the SLHA input string, because it did not match the model parameters that did come out of `.run(...)`.

File: test/test_MSSM_FlexibleDecay.cpp [hunk 1]
```diff
@@ -638,35 +649,35 @@ Block MSOFT Q= 8.61574711E+02
...
+   MSSM_slha m(input);
+   slha_io.fill(m);
+   m.calculate_DRbar_masses();
 
    // -----------------------------------------------------
    // decays with higher-order SM corrections
 
    MSSM_decays decays_with_HO(m, qedqcd, physical_input, flexibledecay_settings);
 
+   const double eps = 1e-8;
```
Comment (wkotlarski) [inline test/test_MSSM_FlexibleDecay.cpp:661]: I must say I don't like such a low precision of comparison.

**Comment by Expander** [comment]
The reason for this low precision is just due to the difference between the model parameters (in the model object) and the (limited precision) SLHA string format.

**Comment by Expander** [comment]
One comment regarding the precision: In your original version the (double precision) model parameters had ~15 digits precision. The SLHA parametes have only 9 digits precision, where the last digit comes from a rounding. So `eps = 1e-8` is a reasonable deviation to expect due to this change.

**Comment by wkotlarski** [comment]
When using only input parameters, the last 16-8 digits would get filled with 0, right? So full double precision comparison still made sense. I'm worried that 1e-8 is too low to spot a lot of changes that might happen in the decays.

**Comment by Expander** [comment]
> When using only input parameters, the last 16-8 digits would get filled with 0, right?

Yes, correct.

> So full double precision comparison still made sense. I'm worried that 1e-8 is too low to spot a lot of changes that might happen in the decays.

I see your point and your are right that we must be careful with the tests! However, we are using `BOOST_CHECK_CLOSE_FRACTION`, which checks for relative difference, not absolute difference. So, if the decays are numerically very small, a check for `1e-8` would still check for `1e-6`% relative difference between these (although small) numbers. You may check this here:

https://godbolt.org/z/31P3sTncr

**Comment by Expander** [comment]
Or are you concerned about more than `1e-8`, i.e. more than 8 digits of each output? Is that what you meant with "too low" precision? If so, we could simply update the numbers.

**Comment by wkotlarski** [comment]
I'm sorry @Expander that it took me so long to reply. I'm on a Dienstraise in Lisbon + bad, awful things happened (so, you know, a week as always). Yes, I'm worried about relative precision. For me 1e-8 seems to low. I'm afraid that some small changes to the code could be misssed by such a "low" comparison. For most things we were using between 1e-11 and 1e-15.

**Comment by Expander** [comment]
No worries!

Then I'd suggest to keep the original precision and simply update the numbers for the decays. I believe we can do this here, because in this PR nothing has changed, except the input parameters (which translates to a change in the output). Would that be ok?

**Comment by wkotlarski** [comment]
Thanks! So this is what i'll do today: I'll keep original precisions of FlexibleDecay tests as they are on the dev branch but I'll go through all the tests (in all models) and update the values we test against.

**Comment by Expander** [comment]
Are you going to do this change on this branch? (Please feel free to do so.)
And will you use the approach proposed in this PR, i.e.
~~~.cpp
   MSSM_slha m(input);
   slha_io.fill(m);
   m.calculate_DRbar_masses();
~~~
This would be best, because then these tests will not break when we change some FS internals, as in the MW treaks branch.

**Comment by wkotlarski** [comment]
@Expander, I'm confused. Why did your change broke only the MSSM FD tests and not, for example, MRSSM ones?

**Comment by Expander** [comment]
I have not tested the decays in the MRSSM, i.e. they may be broken, too.
The reason is that I've disabled the decays at some point because I needed to frequently re-generate the code and wanted things to speed up. :)

**Comment by wkotlarski** [comment]
MRSSM seems to work fine... Which is what confuses me.

**Comment by Expander** [comment]
Do you mean the MRSSM decays test in the branch `feature-MW-tweaks` passes? That's strange, indeed.
It could be that the value of MW does not change much for the tested point, but this is just a guess.

**Comment by wkotlarski** [comment]
No, sorry. Ones on `bugfix-test-MSSM-read-SLHA`. Right, I should check `feature-MW-tweaks`.

**Comment by wkotlarski** [comment]
I think we can close this PR.

**Pull request closed without merging**